powertools-sqs/src/main/java/software/amazon/lambda/powertools/sqs/SqsBatch.java

```diff
@@ -32,6 +32,19 @@
  * {@link SqsBatch#suppressException()} to true. By default its value is false
  * </p>
  *
+ * <p>
+ * If you want certain exceptions to be treated as permanent failures, i.e. exceptions which are not worth retrying and
+ * want such message should be moved to configured dead letter queue of the source SQS queue, you can use
+ * {@link SqsBatch#nonRetryableExceptions()} to configure such exceptions.
+ * If you want such messages to be deleted instead, set {@link SqsBatch#deleteNonRetryableMessageFromQueue()} to true.
+ * By default its value is false.
+ *
+ * If there is no DLQ configured on source SQS queue and {@link SqsBatch#nonRetryableExceptions()} attribute is set, if
+ * nonRetryableExceptions occurs from {@link SqsMessageHandler}, such exceptions will still be treated as temporary
+ * exceptions and the message will be move back to source SQS queue for reprocessing. Same behaviour occurs if for some
+ * reason utility is unable to move message to the DLQ. This can occur because of missing permissions.
```

Comment:
What are the required permissions?

Reply:
I meant to document that as part of documentation update